python/simulators/PyMain.hpp

```diff
@@ -34,6 +34,22 @@ class PyMain : public Main
 public:
     using Main::Main;
 
+    void setArguments(const std::vector<std::string>& args)
+    {
+        if (args.empty()) {
+            return;
+        }
+
+        this->argc_ = 2 + args.size();
```

Comment:
I think the logic here is correct, but it would be nice with a comment explaining it. It is hard to juggle `args`, `argv_` and `argv_python_` without getting confused.

Reply:
tried to add an explanation.